### ISSUE TITLE
USERS - READ (VIEW USER DETAILS AND VIEW USER LIST) 

### ISSUE THREAD

**Issue by keanacobarde** [nwelto/rare#18]
# User Story
- The user/frontend developer should be able to obtain all user details, when given a specific UserId.
- The user should be able to see ALL users.

# Acceptance Criteria
- Two endpoints will be created within this ticket, both of which are GET endpoints but accomplish different tasks:
GET ALL USERS: "/api/users", should display ALL users. 
GET USER BY ID: "/api/users/{id}", should display only the details of a specific user 

# Dependecies
- #2 

# Dev Notes
GET ALL USERS, would most likely be written like the following: 
```
app.MapGet("/api/employees", () =>
{
    return employees;
});
```

GET USERS BY ID:
```
app.MapGet("/api/servicetickets/{id}", (int id) =>
{
    ServiceTicket serviceTicket = serviceTickets.FirstOrDefault(st => st.Id == id);
    if (serviceTicket == null)
    {
        return Results.NotFound();
    }
    serviceTicket.Employee = employees.FirstOrDefault(e => e.Id == serviceTicket.EmployeeId);
    return Results.Ok(serviceTicket);
});
```

- Test data may need to be created for the testing of this data. Utilize sources like ChatGPT or Mockaroo to obtain dummy data.

**Comment by keanacobarde** [comment]
- The following frontend implementations will utilize the data call from this same endpoint:
> Filter by User: The frontend promise will call for all users, the backend will use the GET ALL USERS endpoint to supply data to the frontend. All filtering is done through the UI.
- Testing will be done through PostMan and Swagger

**Comment by keanacobarde** [comment]
CLOSED PER PR #27
